numpy/core/arrayprint.py

```diff
@@ -272,22 +272,24 @@ def get_printoptions():
     """
     return _format_options.copy()
 
-def _leading_trailing(a):
+
+def _leading_trailing(a, index=()):
+    """
+    Keep only the N-D corners (leading and trailing edges) of an array
+    """
     edgeitems =  _format_options['edgeitems']
-    if a.ndim == 1:
-        if len(a) > 2*edgeitems:
-            b = concatenate((a[:edgeitems], a[-edgeitems:]))
-        else:
-            b = a
+    axis = len(index)
+    if axis == a.ndim:
+        return a[index]
+
+    if a.shape[axis] > 2*edgeitems:
+        return concatenate((
+            _leading_trailing(a, index + np.s_[ :edgeitems,]),
```

Comment:
That `,` at the end really threw me of: I'd either use `np.index_exp[:edgeitems]` (which always returns a tuple), or go for the, for me, more natural `(slice(None, edgeitems),)`

Reply:
`s_` is slightly faster, but I guess we don't care. I'll switch to `index_exp`